Task-56521: Fix duplication of news and activity items in favorites drawer

SaraBoutej opened the pull request:
Before this Fix when we bookmark a news, we have the related activity that is bookmarked too, so in the drawer of recent favorites we have duplication of items: activity + news.
With this fix, we add a boolean property "hidden" when bookmark an item to avoid the duplication of an item in the favorites drawer.

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

hakermi committed:
TASK-56601: Duplication and bad detection and display of tags (#1434)

Prior to this changes, when creation a tag and then press the space button each time will create another tag with number of spaces converted to &nbsp; as the matcher of tags pattern will consider it as a normal character
follows the tag text wich will duplicate the tag depends on the number of spaces that follows it and display it with the &nbsp; in the unified search when selecting tags, this because of the returned html content of the
activity when searching for tags, This PR will sanitize the content to correctly unescape the &nbsp; and replacing it by spaces to allow the tags pattern matcher to work as expected.

SaraBoutej committed:
Task-56521: Fix duplication of news and activity items in favorites drawer

ghost commented:
### CodeSee Review Map:
[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/Meeds-io/social/1437/3fca4fc2/41e87a2377f873b99d69f41b4bf634214f0b6c74.svg)](https://app.codesee.io/r/reviews?pr=1437&src=https%3A%2F%2Fgithub.com%2FMeeds-io%2Fsocial)

Review [in an interactive map](https://app.codesee.io/r/reviews?pr=1437&src=https%3A%2F%2Fgithub.com%2FMeeds-io%2Fsocial)

View [more CodeSee Maps](https://app.codesee.io/r/maps?src=https%3A%2F%2Fgithub.com%2FMeeds-io%2Fsocial)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

hakermi committed:
TASK-56601: Tags are duplicated when they are ending with white-spaces (#1439)

Prior to this change, when creating a tag and then add a whitespace right after it it will be recognized as a new tag and then duplicated, this due to the fact that the &nbsp; char are converted to and space
and recognized in java pattern as a char NBSP and so will be considered as a part fo the tagtext.
This PR should remove this weird char from the tag text to avoid duplicating it

hakermi committed:
TASK-56832: Unescaped html characters in favorite activity titles (#1442)

Prior to this change the html content of the activity title are displayed  without escaping the special html chars,
This PR uses the custom html-sanitized directive to correctly unescape these chars

azayati committed:
Task-55805: [Accessibility - Contrast - 3.2] Change the default primary color (#1441)

azayati committed:
Task-55805: [Accessibility - Contrast - 3.2] Change the default primary color (#1441)

sofyenne committed:
Task-56602: Fix icons misalignment in Toolbar. (#1436)

* Task-56602: Copy of [AUTH][UXFT] TopBar_BUG01.2: Favorite icon alignment

problem: bad alignement of icon favorite

* Update webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ExoTopBarFavorites.vue

Co-authored-by: Boubaker Khanfir <boubaker.khanfir@exoplatform.com>

Co-authored-by: Boubaker Khanfir <boubaker.khanfir@exoplatform.com>

sofyenne committed:
Task-56602: Fix icons misalignment in Toolbar. (#1436)

* Task-56602: Copy of [AUTH][UXFT] TopBar_BUG01.2: Favorite icon alignment

problem: bad alignement of icon favorite

* Update webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ExoTopBarFavorites.vue

Co-authored-by: Boubaker Khanfir <boubaker.khanfir@exoplatform.com>

Co-authored-by: Boubaker Khanfir <boubaker.khanfir@exoplatform.com>

hakermi committed:
TASK-56851: Add and ellipsis to space name in space Access portlet when the space name is very long (#1443)

Prior to this change, On mobile screen, when the space where the user has been invited has a long name then it's not well displayed in the space access page and the name overflows the spaceblock access area.
THis Pr should add the class text-truncate after adding it to the dedicated portlet style in platform-ui to truncate the name and display an ellipsis with a html tooltip

boubaker commented on webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/footer/actions/ActivityFavoriteAction.vue:
```diff
@@ -60,6 +61,9 @@ export default {
       return this.activityTypeExtension && this.activityTypeExtension.name;
     },
+    hiddenFavorite() {
+      return this.activity && this.activity.type && this.activity.type === 'news' ? true : false;
```
shouldn't embed eXo addon name inside Meeds code. Meeds code should be flexible to allow the addon to parameter this kind of properties through extensionRegistry

boubaker commented on component/core/src/main/java/org/exoplatform/social/core/metadata/favorite/FavoriteServiceImpl.java:
```diff
@@ -39,15 +40,13 @@ public FavoriteServiceImpl(MetadataService metadataService) {
 
   @Override
-  public void createFavorite(Favorite favorite) throws ObjectAlreadyExistsException {
+  public void createFavorite(Favorite favorite, Map<String, String> favoriteProperties) throws ObjectAlreadyExistsException {
```
```suggestion
  @Override
  public void createFavorite(Favorite favorite) throws ObjectAlreadyExistsException {
    this.createFavorite(favorite, false);
  }

  @Override
  public void createFavorite(Favorite favorite, boolean hiddeInList) throws ObjectAlreadyExistsException {
```
In term of API definition, multiple things:
1/ It's better to keep the original operation with simple input without options `createFavorite(Favorite favorite)`. This will allow to use the API for simple use cases without complexity.
2/ It's better to not add a generic `properties` Map while our use case is simple: whether to hide favorites while listing or not.
3/ You added an option for favorite creation, but it's not clear (in term of API perspective) where it will be used. In fact, a new method to list favorites should be added in Service layer to allow filter by visibility.

SaraBoutej committed:
Merge branch 'feature/maintenance' into Task-56521
